module-bsp/board/rt1051/drivers/RT1051DriverI2C.cpp

```diff
@@ -76,13 +77,19 @@ namespace drivers
         cpp_freertos::LockGuard lock(mutex);
         auto ret = BOARD_LPI2C_Receive(base, addr.deviceAddress, addr.subAddress, addr.subAddressSize, rxBuff, size);
         if (ret != kStatus_Success) {
+            lastError = static_cast<int32_t>(ret);
             return -1; // TODO:M.P: fix me
         }
         else {
             return size;
         }
     }
 
+    int32_t RT1051DriverI2C::GetLastError()
+    {
+        return lastError;
```

Comment:
Could you please explain what the purpose of this method is? Wouldn't it be better to just return a correct retu code and then handle it later?

Reply:
I agree that GetLastError function isn't the best solution here. I added proper error codes for I2C and a function that translate generic error codes to I2C specific.

Reply:
Done